weavelet.go

```diff
@@ -234,24 +234,9 @@ func (w *weavelet) start() (Instance, error) {
 			}
 		}
 
-		// Start serving the transport.
-		serve := func(lis net.Listener, transport *transport) {
-			if lis == nil || transport == nil {
-				// TODO(spetrovic): Can this ever happen?
-				return
-			}
```

Comment:
I looked into this TODO. `lis` is initialized in the constructor to `WeaveletConn`

https://github.com/ServiceWeaver/weaver/blob/65037e069c1fbce40887617eec7a167f463efda6/internal/envelope/conn/weavelet_conn.go#L94

And `transport` is initialized in `newWeavelet`

https://github.com/ServiceWeaver/weaver/blob/65037e069c1fbce40887617eec7a167f463efda6/weavelet.go#L150

Which I think means neither of these can be nil in practice, so I removed the TODO and these checks for nil.